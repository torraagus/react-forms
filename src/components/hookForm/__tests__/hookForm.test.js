import React from "react";
import ReactDOM from "react-dom";

import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import "@testing-library/jest-dom/extend-expect";

import HookForm from "../hookForm.component";

beforeEach(cleanup);

it("should render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<HookForm />, div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<HookForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("if form was correctly submited when no errors are present", async () => {
  const onSubmit = jest.fn();
  const { getByTestId } = render(<HookForm onSubmit={onSubmit} />);
  await fireEvent.change(getByTestId("example"), { target: { value: "test" } });
  expect(getByTestId("example").value).toBe("test");
  await fireEvent.submit(getByTestId("form"));
});

test("if form was correctly submited when some errors are present", async () => {
  const { getByTestId } = render(<HookForm />);
  await fireEvent.submit(getByTestId("form"));
  expect(getByTestId("error").textContent).toBe("This field is required");
});
