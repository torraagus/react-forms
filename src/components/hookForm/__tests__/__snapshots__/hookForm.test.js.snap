// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form
    data-testid="form"
    onSubmit={[Function]}
  >
    <input
      data-testid="username"
      name="username"
      placeholder="Username..."
    />
    <br />
    <input
      data-testid="password"
      name="password"
      placeholder="Password..."
      type="password"
    />
    <br />
    <br />
    <input
      type="submit"
      value="Send"
    />
  </form>
</div>
`;
